### PR TITLE
REF: upgrade eslint, prettier, fix errors, reformat everything 💄

### PR DESCRIPTION
**Pull request by limpbrains**
- prettier upgraded to v2. `"arrowParens": "avoid"` in config to reduce the diff 
- package.json simplified, added missing files to eslint
- eslint plugins and config files upgraded to the latest versions
- No more formatting conflicts between eslint and prettier
- `react-native/no-inline-styles` enabled everywhere, except BlueComponents
- `obj.hasOwnProperty(prop)` replaced with `prop in obj` to fix eslint error
- a few more minor eslint errors fixed 
- reviewed all `eslint-disable` rules, removed unused. 
- eslint --fix and prettier reformat everything

_PS
I've also tried to replace `require` and `module.exports` with `import` and `export` es6, but failed, too much to fix :)
Maybe next time_

### CODE REVIEW

**Comment by limpbrains** [comment]
I can't understand why tests have failed from the logs
When I'm running them locally on master branch, they fail with Timeout error

**Comment by Overtorment** [comment]
`txdatas were coming back null from BlueElectrum because of high batchsize` - this is just too heavy address to fetch, Ill probably mute this test for now.
the rest Ill try to debug later

**Comment by limpbrains** [comment]
> `txdatas were coming back null from BlueElectrum because of high batchsize` - this is just too heavy address to fetch, Ill probably mute this test for now.
> the rest Ill try to debug later

I've found `ReferenceError: You are trying to import a file after the Jest environment has been torn down.` error in logs, so looks like my problem is unrelated to yours. Probably eslint swapped some lines in tests or in BlueElectrum.

**Comment by limpbrains** [comment]
> `txdatas were coming back null from BlueElectrum because of high batchsize` - this is just too heavy address to fetch, Ill probably mute this test for now.
> the rest Ill try to debug later

You were right it is because `txdatas were coming back null from BlueElectrum because of high batchsize`

Also, now it failed because connection error to `electrum2.bluewallet.io`
@Overtorment can you please take a look, I don't know what to do with this PR? Is it wrong or tests are wrong?

File: screen/wallets/details.js [hunk 1]
```diff
@@ -278,10 +278,10 @@ export default class WalletDetails extends Component {
                           onValueChange={value => this.onUseWithHardwareWalletSwitch(value)}
                         />
                       </View>
-                      <React.Fragment>
+                      <>
```
Comment (Overtorment) [inline screen/wallets/details.js:281]: wow is that legit?

**Comment by Overtorment** [comment]
btw lets resubmit pr as a branch in this repo. will run more tests & builds

File: BlueApp.js [hunk 2]
```diff
@@ -3,11 +3,11 @@
...
-let prompt = require('./prompt');
-let EV = require('./events');
-let currency = require('./currency');
-let loc = require('./loc');
-let BlueElectrum = require('./BlueElectrum'); // eslint-disable-line
+const prompt = require('./prompt');
+const EV = require('./events');
+const currency = require('./currency');
+const loc = require('./loc');
+require('./BlueElectrum');
```
[the remark this answers is not in the document]
Comment (limpbrains) [reply BlueApp.js]: Omg, it works, amazing

File: screen/wallets/details.js (hunk 1 above)
Comment (limpbrains) [reply screen/wallets/details.js:281]: Yeah

**Pull request closed without merging**